test/lit.cfg

```diff
@@ -324,13 +324,14 @@ if test_sdk_overlay_dir is not None:
 test_options = os.environ.get('SWIFT_TEST_OPTIONS')
 config.swift_test_options = ' ' + test_options if test_options else ''
 
-clang_module_cache_path = tempfile.mkdtemp(prefix="swift-testsuite-clang-module-cache")
+lit_config.note("Using test results dir: " + config.swift_test_results_dir)
+
+clang_module_cache_path = os.path.join(config.swift_test_results_dir, 'ModuleCache')
```

Comment:
It was really important for this directory to be an empty one, otherwise we had issues with stale module caches.  Same with completion cache.